### PR TITLE
Dynamic variable

### PR DESCRIPTION
**Pull request by jhlee-mitre**
# Summary
Dynamic variable

## New behavior
Users can set up dynamic variable as a placeholder for user-defined execution time values for [TestScript.variable](https://www.hl7.org/fhir/testscript-definitions.html#TestScript.variable). Technically dynamic variable will replace value in [defaultValue](https://www.hl7.org/fhir/testscript-definitions.html#TestScript.variable.defaultValue) at the beginning of test execution. 
There are two ways to set up dynamic variable:

### Config.yml
variable : [ARRAY] 

Use format 
```
- name1=value1
- name2=value2
```

### Command line argument
bundle exec testscript_engine execution --variable [ARRAY] 

Use format 
`name1=value1 name2=value2`

Use quotations if there are spaces.

## Testing guidance
Sample TestScript `testscript_dynamic_variable` is included. 
Use the examples below from command line argument.

`bundle exec testscript_engine execute --testscript_path "/spec/examples/testscript_unittest" --variable PatientResourceId1=var1changed PatientResourceId2=var2changed --testscript_name testscript_dynamic_variable`

### Expected result

```
STARTING TO TEST
         GETTING: http://server.fire.ly/Patient/var1changed
         (✓) Executed Operation: [unlabeled]
         (✗) Response: Expected Response equals ["okay"], but found [].
         GETTING: http://server.fire.ly/Patient/var2changed
         (✓) Executed Operation: [unlabeled]
         (✗) Response: Expected Response equals ["okay"], but found [].
      FINISHED TEST.

### CODE REVIEW

**Review by karlnaden — APPROVED** [review]
Looks like it works as needed. One small documentation fix and a potential idea for improving whitespace handling, but probably not one to spend time on.

File: lib/testscript_engine.rb [hunk 1]
```diff
@@ -93,19 +94,31 @@ def load_input
...
       end
     end
   rescue StandardError
     error(:unable_to_create_runnable, script.name)
   end
 
+  def dynamic_variable(script)
+    script.variable.each do |v|
+      variable.each do |v2|
+        v.defaultValue = v2.gsub(" =","=").gsub("= ","=").split("=").last if v2.gsub(" =","=").gsub("= ","=").split("=").first == v.name && v.defaultValue != nil
```
[the remark this answers is not in the document]
Comment (jhlee-mitre) [reply lib/testscript_engine.rb:116]: Yep, this is undergraduate coding. I am not really good at regular expression in Ruby. I will take some time to convert it shorter.

Thor seems to not provide much flexibility as to change its syntax and space is the most tricky thing especially when to deal with multiple arguments. I somehow inclined to direct users to use config.yml for any complicated settings and use CLI arguments for adhoc / self-service testing as necessary. But it's hard and/or early to tell at this point. Just my thoughts.

[the remark this answers is not in the document]
Comment (jhlee-mitre) [reply lib/testscript_engine.rb:116]: Continuing white space, do we assume possibility of white space in value in variable? 
I assumed Yes. 
If not, removing white spaces will be easy fix without ugly coding.

[the remark this answers is not in the document]
Comment (karlnaden) [reply lib/testscript_engine.rb:116]: I don't think removing all white spaces makes sense. I would either:
- leave as is
- make simpler and assume that there are no white spaces around the equals sign

**Pull request merged**